File: gmr-app/src/models/dial-detection-config.ts
export interface DialDetectionConfig {
  method: DetectionMethodType;

  adaptiveThreshold: boolean;
  morphology: boolean;

  cannyEdgeThreshold1: number;
  cannyEdgeThreshold2: number;

  blurKernelSize: number;
  thresholdValue: number;


  edgeDetectionThreshold: number;
  contourCircularThreshold: number;
}

export interface HoughCircleConfig {
  dp: number;
  minDist: number;
  param1: number;
  param2: number;
  minRadius: number;
  maxRadius: number;
}

export type DetectionMethodType = 'COMBINED' | 'HOUGH' | 'CONTOUR' | 'EDGE'

File: gmr-api/src/main/java/au/com/dobotics/gmr/pipeline/config/DialDetectionConfig.java
package au.com.dobotics.gmr.pipeline.config;

import au.com.dobotics.gmr.pipeline.DetectionMethodType;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Builder
@Data
@NoArgsConstructor
@AllArgsConstructor
@JsonIgnoreProperties(ignoreUnknown = true)
public class DialDetectionConfig {

    private DetectionMethodType method;

    private boolean adaptiveThreshold;
    private boolean morphology;

    private double cannyEdgeThreshold1;           // Canny edge threshold
    private double cannyEdgeThreshold2;            // Accumulator threshold

    private int blurKernelSize;             // Gaussian blur kernel size
    private int thresholdValue;          // Binary threshold value

    private double edgeDetectionThreshold;
    private double contourCircularThreshold;

}
